packages/plugin-legacy/src/index.ts

```diff
@@ -752,12 +754,11 @@ function isLegacyBundle(
 function recordAndRemovePolyfillBabelPlugin(
   polyfills: Set<string>,
 ): BabelPlugin {
-  return ({ types: t }): BabelPlugin => ({
+  return ({ types: t }: { types: typeof BabelTypes }): BabelPlugin => ({
     name: 'vite-remove-polyfill-import',
     post({ path }) {
       path.get('body').forEach((p) => {
-        if (t.isImportDeclaration(p)) {
-          // @ts-expect-error
+        if (t.isImportDeclaration(p.node)) {
```

Comment:
What is the reason for this change?

Reply:
`t.isImportDeclaration` only expects `Node` to be passed to the argument. `p` is `NodePath` and `p.node` is `Node`.
The previous code also works because `NodePath` and `Node` both has `type` property.
https://github.com/babel/babel/blob/df733b18ae88f370caddecc30c6d96844007c411/packages/babel-types/src/validators/generated/index.ts#L1062-L1078
But given that `t.isImportDeclaration` narrows the argument to `ImportDeclaration` instead of `NodePath<ImportDeclaration>`, passing `p.node` is more correct.

We could use `p.isImportDeclaration()` instead of `t.isImportDeclaration(p.node)`, too.